### PR TITLE
CMake: Default to cmake enabled for x86 linux

### PR DESCRIPTION
**Pull request by dnakamura**
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
I'm concerned about https://github.com/eclipse/openj9/issues/9126, although it doesn't seem to be occurring often any more. Was anything done to resolve this?

**Comment by pshipton** [comment]
We should also confirm the special.system tests are passing. The internal testing is failing although not related to cmake. I attempted a run on the weekend, but didn't do it correctly so I'll have to try again. Similarly the XL build.

I'll start some new builds shortly.

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Pipeline-Release-Build/218/

**Comment by pshipton** [comment]
special.system testing on the xlinux cmake build from last night
https://ci.eclipse.org/openj9/view/Test/job/Grinder/913

**Comment by dnakamura** [comment]
> I'm concerned about [eclipse/openj9#9126](https://github.com/eclipse/openj9/issues/9126), although it doesn't seem to be occurring often any more. Was anything done to resolve this?

Except for some early cases, the issue seems to be limited to jdk8. In addition, I believe the issue was resolved with eclipse/omr#5266, but given the intermittent nature of the issue its impossible to say for sure without seeing if it pops up again

File: closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -59,11 +59,14 @@ AC_DEFUN([OPENJ9_CONFIGURE_CMAKE],
       fi
     ],
     [
-      if test "x$OPENJDK_BUILD_OS" = xmacosx ; then
-        with_cmake=cmake
-      else
-        with_cmake=no
-      fi
+      case "$OPENJ9_PLATFORM_CODE" in
+        xa64|oa64)
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4:63]: Please put these in alphabetical order (`oa64|xa64`) like on [line 241](https://github.com/dnakamura/openj9-openjdk-jdk11/blob/cmake_xlinux/closed/autoconf/custom-hook.m4#L241).

**Comment by keithc-ca** [comment]
Jenkins copyright check

**Pull request merged**